src/platform/w3dengine/client/w3ddisplay.cpp

```diff
@@ -2155,13 +2161,23 @@ void Reset_D3D_Device(bool restore_assets)
                         true,
                         true,
                         true);
-                    OSVERSIONINFOA VersionInformation;
 
-                    if (GetVersionEx(&VersionInformation)) {
+#if USE_LEGACY_RESET_D3D_DEVICE
+                    // Original code uses deprecated function.
+                    OSVERSIONINFOA VersionInformation;
+                    ZeroMemory(&VersionInformation, sizeof(OSVERSIONINFOA));
+                    VersionInformation.dwOSVersionInfoSize = sizeof(OSVERSIONINFOA);
+                    if (GetVersionExA(&VersionInformation)) {
                         if (VersionInformation.dwPlatformId == VER_PLATFORM_WIN32_WINDOWS) {
                             W3D::Invalidate_Textures();
                         }
                     }
+#else
+                    // Replacement code uses supported Windows API. Available with Windows 2000 and higher.
+                    if (!IsWindowsXPOrGreater()) {
```

Comment:
Actually, this is stupid, because it would not compile on earlier windows to begin with.

Reply:
Fixed.